fractal_server/app/models/workflow.py

```diff
@@ -139,12 +141,19 @@ async def insert_task(
             db: TBD
             commit: TBD
         """
+        if self.id is None:
+            raise ValueError(
+                "Cannot use `insert_task` for a non-committed Workflow"
```

Comment:
I suggest we go with an even simpler message (independent on whether the workflow was committed or maybe something else went wrong).
Example: "Cannot use `Workflow.insert_task` when `Workflow.id=None`"